File: exec/transport.lua
Ne = 100
Nangles = 32
xb = 10
fe_order = 2

niter = 50 

function Sigma_t(x) 
	return math.cos(math.pi*x*2/xb) + 1
end 

function Sigma_s(x)
	return .7*Sigma_t(x)
end 

function Source(x, mu) 
	if (x > 4 and x < 6) then 
		return 1 
	else 
		return 0
	end
end 

function Inflow(x, mu) 
	if (mu > 0) then 
		return 1
	else 
		return 0
	end
end

File: exec/transport.cpp
#include "TRT.hpp"

using namespace trt; 
using namespace std; 

LuaReader lua; 

double Sigma_t(double x) {
	return lua.ScalarFunction("Sigma_t", x); 
}
double Sigma_s(double x) {
	return lua.ScalarFunction("Sigma_s", x); 
}
double Source(double x, double mu) {
	return lua.ScalarFunction("Source", x, mu); 
}
double Inflow(double x, double mu) {
	return lua.ScalarFunction("Inflow", x, mu); 
}

int main(int argc, char* argv[]) {
	lua.Parse(argc, argv, "transport.lua");

	int Ne = lua.Int("Ne"); 
	int Nangles = lua.Int("Nangles"); 
	int p = lua.FEOrder();  
	double xb = lua.Double("xb"); 

	FunctionOpacity sig_t(Sigma_t); 
	FunctionOpacity sig_s(Sigma_s); 
	FunctionStateCoefficient source(Source); 
	FunctionStateCoefficient inflow(Inflow); 

	L2Space l2(Ne, xb, p); 
	TransportOperator transport(&l2, Nangles, &sig_s, &sig_t, &source, &inflow); 

	TVector psi(&l2, Nangles); 
	transport.SourceIteration(psi, lua.Int("niter", 25), lua.Double("tol", 1e-6));
	GridFunction phi(&l2); 
	transport.ComputeScalarFlux(psi, phi); 

	Writer writer; 
	writer.Add(phi, "phi"); 
	writer.Write(); 
}